nix-shell.el

```diff
@@ -152,6 +154,13 @@ The DRV file to use."
 	;; (setq-local process-environment nil)
 	)
 
+      ;; Set the LD_LIBRARY_PATH where applicable
+      (when ld-library-path
+	(make-local-variable 'process-environment)
+	(setq process-environment
+	      (cons (format "LD_LIBRARY_PATH=%s" ld-library-path)
```

Comment:
can we concat this with the value of `(getenv "LD_LIBRARY_PATH")`? that way you're not overriding any user settings.

Reply:
Sure - I wasn't sure what was more nix-like (since I wanted to mimic the same behaviour as if I had done 'nix-shell' in the REPL).